chart/fediblockhole.conf.toml

```diff
@@ -0,0 +1,67 @@
+# List of instances to read blocklists from.
+# If the instance makes its blocklist public, no authorization token is needed.
+#   Otherwise, `token` is a Bearer token authorised to read domain_blocks.
+# If `admin` = True, use the more detailed admin API, which requires a token with a 
+#   higher level of authorization.
+# If `import_fields` are provided, only import these fields from the instance.
+#   Overrides the global `import_fields` setting.
+blocklist_instance_sources = [
+  # { domain = 'public.blocklist'}, # an instance with a public list of domain_blocks
+  # { domain = 'jorts.horse', token = '<a_different_token>' }, # user accessible block list
+  # { domain = 'eigenmagic.net', token = '<a_token_with_read_auth>', admin = true }, # admin access required
+]
```

Comment:
Having a copy of the sample `fediblockhole.conf.toml` file here will require making the same changes in more than one place if the sample gets changed. Is there a way to make this easier to maintain?

Reply:
I *think* the chart needs a dummy file to create the part that pulls the value map to create the conf.toml file in /etc/default. When you deploy your own kustomization, it will actually pull the local configmap in your own flux repo. Open to ideas as to how to avoid dealing with changes in the sample file - that had occurred to me, but I thought we could deal with that when we rebuild the container each time the version increments?

Reply:
I don't think this part of the template works unless there is a file for it to pull - but it may not be necessary for deployment at all...

chart/templates/configmap-conf-toml.yaml

apiVersion: v1
kind: ConfigMap
metadata:
  name: {{ include "fediblockhole.fullname" . }}-conf-toml
  labels:
    {{- include "fediblockhole.labels" . | nindent 4 }}
data:
  {{ (.Files.Glob "fediblockhole.conf.toml").AsConfig | nindent 4 }}

Reply:
I just know that people will blindly use the sample provided if there is one. If it isn't maintained, it'll break stuff and they'll open bug reports when it does. Let's make it easy for people to get it right and hard to get it wrong, and really obvious when they do get it wrong _why_ it's wrong and how to fix it themselves.

Reply:
Let's remove that .yaml file from the template directory and see if it still deploys. If it does, I'd be in favor of removing it and placing an expanded version in the samples directory as you suggest. That's what the flux kustomization ends up using anyway, like this:

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: fediblockhole-conf-toml
  namespace: fediblockhole
data:
  fediblockhole.conf.toml: |-  
    # List of instances to read blocklists from.
    # If the instance makes its blocklist public, no authorization token is needed.
    #   Otherwise, `token` is a Bearer token authorised to read domain_blocks.
    # If `admin` = True, use the more detailed admin API, which requires a token with a 
    #   higher level of authorization.
    # If `import_fields` are provided, only import these fields from the instance.
    #   Overrides the global `import_fields` setting.
    blocklist_instance_sources = [
      # { domain = 'public.blocklist'}, # an instance with a public list of domain_blocks
      # { domain = 'jorts.horse', token = '<a_different_token>' }, # user accessible block list
      # { domain = 'eigenmagic.net', token = '<a_token_with_read_auth>', admin = true }, # admin access required
    ]
    ...
```